prusti-tests/tests/verify/pass/time-resources/array_cmp.rs

```diff
@@ -0,0 +1,34 @@
+// compile-flags: -Ptime_reasoning=true
+
+use prusti_contracts::*;
+
+#[requires(a.len() == b.len())]
+#[requires(time_credits(a.len() + 1))]
+#[ensures(time_receipts(a.len() + 1))]
+fn safe_array_cmp<T: Eq>(a: &[T], b: &[T]) -> bool {
+  let mut i = 0;
+  let mut res = true;
+  while i < a.len() {
```

Comment:
Why don't we need to put `time_credits` in the invariant?

Reply:
Prusti didn't complain so I thought I didn't have to and the `loop.rs` tests were passing without so I thought that it was okay but playing a bit with the generated Viper code showed me that it doesn't 😅